software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,656 @@
+#
+#   {manycore_dim_y}  Mesh Y dimension of manycore 
+#   {manycore_dim_x}  Mesh X dimension of manycore 
+#   {per_tile}        Generate separate stats file for each tile default = False
+#   {input}           Vanilla stats input file     default = vanilla_stats.log
+
+
+
+import sys
+import argparse
+import os
+import re
+import csv
+from enum import Enum
+from collections import Counter
+
+
+
+# Default coordinates of origin tile
+BSG_ORG_X = 0
+BSG_ORG_Y = 1
+
+# Default input values
+DEFAULT_INPUT_FILE = "vanilla_stats.csv"
+
+
+
+
+
+# These values are used by the manycore library in bsg_print_stat instructions
+# they are added to the tag value to determine the tile group that triggered the stat
+# and also the type of stat (stand-alone stat, start, or end)
+# the value of these paramters should match their counterpart inside 
+# bsg_manycore/software/bsg_manycore_lib/bsg_manycore.h
+# For formatting, see the CudaStatTag class
+BSG_STAT_TAG_WIDTH   = 8
+BSG_STAT_TG_ID_WIDTH = 10
+BSG_STAT_X_WIDTH     = 6
+BSG_STAT_Y_WIDTH     = 6
+BSG_STAT_TYPE_WIDTH  = 2
+BSG_STAT_TAG_MASK   = ((1 << BSG_STAT_TAG_WIDTH) - 1)
+BSG_STAT_TG_ID_MASK = ((1 << BSG_STAT_TG_ID_WIDTH) - 1)
+BSG_STAT_X_MASK     = ((1 << BSG_STAT_X_WIDTH) - 1)
+BSG_STAT_Y_MASK     = ((1 << BSG_STAT_Y_WIDTH) - 1)
+BSG_STAT_TYPE_STAT  = 0
+BSG_STAT_TYPE_START = 1
+BSG_STAT_TYPE_END   = 2
+
+
+# CudaStatTag class
+# Is instantiated by a packet tag value that is recieved from a 
+# bsg_cuda_print_stat(tag) insruction
+# Breaks down the tag into (type, y, x, tg_id, tag>
+# type of tag could be start, end, stat
+# x,y are coordinates of the tile that triggered the print_stat instruciton
+# tg_id is the tile group id of the tile that triggered the print_stat instruction
+# Formatting for bsg_cuda_print_stat instructions
+# Section                 stat type  -   y cord   -   x cord   -    tile group id   -        tag
+# of bits                <----2----> -   <--6-->  -   <--6-->  -   <------10----->  -   <-----8----->
+# Stat type value: {"stat":0, "start":1, "end":2}
+class CudaStatTag:
+  def __init__(self, tag):
+    self.stat_val = tag;
+
+  def get_tag(self):
+    return ((self.stat_val) & BSG_STAT_TAG_MASK)
+  def get_tg_id(self):
+    return ((self.stat_val >> (BSG_STAT_TAG_WIDTH)) & BSG_STAT_TG_ID_MASK)
+  def get_x(self):
+    return ((self.stat_val >> (BSG_STAT_TAG_WIDTH + BSG_STAT_TG_ID_WIDTH)) & BSG_STAT_X_MASK)
+  def get_y(self):
+    return ((self.stat_val >> (BSG_STAT_TAG_WIDTH + BSG_STAT_TG_ID_WIDTH + BSG_STAT_X_WIDTH)) & BSG_STAT_Y_MASK)
+  def get_type(self):
+    return ((self.stat_val >> (BSG_STAT_TAG_WIDTH + BSG_STAT_TG_ID_WIDTH + BSG_STAT_X_WIDTH + BSG_STAT_Y_WIDTH)))
+
+  def isStart(self):
+    return (self.get_type() == BSG_STAT_TYPE_START)
+  def isEnd(self):
+    return (self.get_type() == BSG_STAT_TYPE_END)
+  def isStat(self):
+    return (self.get_type() == BSG_STAT_TYPE_STAT)
```

Comment:
If you really want to go full Python, see the @property decorator:  https://docs.python.org/3/library/functions.html#property

Then you can do cst.tag, csg.tg_id, cst.x, cst.y, cst.isStart, etc.

Reply:
(This isn't necessary, but is a useful feature sometimes)

Reply:
I store your code, thanks! :D